### ISSUE TITLE
Type error is thrown when streaming a request that falsely claims to be gzip encoded

### ISSUE THREAD

**Issue by adriana-saroz** [sindresorhus/got#1456]
#### Describe the bug

 - Node.js version: 12.14
 - OS & version: Ubuntu 18

This error occurs when HTTP request's headers claim that content is gzip encoded while it's not and the response body is empty. 

```
TypeError: Cannot read property 'toString' of undefined
  at /node_modules/got/dist/source/core/index.js:1187:50
  at processTicksAndRejections (internal/process/task_queues.js:94:5)
```

The error was introduced in this commit https://github.com/sindresorhus/got/commit/707219825b2f8ff3df597230a53244636df21642#diff-8eef38c70c64f535c3b8046b5ce40d9eR1664 by moving `response.body = response.rawBody.toString();` outside of the catch clause. 

#### Actual behavior

When streaming an empty response that claims to be gziped there's an unhandled error being thrown:
```
TypeError: Cannot read property 'toString' of undefined
  at /node_modules/got/dist/source/core/index.js:1187:50
  at processTicksAndRejections (internal/process/task_queues.js:94:5)
```

#### Expected behavior

When streaming an empty response that claims to be gziped the expected result is to get a parse error on the stream's error event handler: `Parse Error: Expected HTTP/`. This is the behavior in version 11.5.2

#### Code to reproduce

```
await new Promise((resolve, reject) => {

  const url = 'https://mwi.usma.edu/feed/';

  const stream = got.stream(url, {
    decompress: true
  });

  stream.on('error', reject);

  // stream.on('finish', resolve);

}).catch(err => console.log(err.message));
```

#### Checklist

- [x] I have read the documentation.
- [x] I have tried my code with the latest version of Node.js and Got.

**Comment by szmarczak** [comment]
Can you try 11.6.2 please?

**Comment by adriana-saroz** [comment]
It's fixed in 11.6.2, thank you!
